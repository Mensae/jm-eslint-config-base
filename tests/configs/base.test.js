/* eslint-disable @getify/proper-arrows/this */
/* eslint-disable @getify/proper-arrows/where */
/* eslint-disable @getify/proper-arrows/name */
/* eslint-env jest */

const baseConfig = require('../../configs/base.js');

describe('eslint-config-base', () => {
  it('should match the snapshot', () => {
    expect(baseConfig).toMatchSnapshot();
  });
});
