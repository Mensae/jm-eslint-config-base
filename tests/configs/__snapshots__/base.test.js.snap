// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-base should match the snapshot 1`] = `
{
  "env": {
    "es2022": true,
  },
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
  },
  "plugins": [
    "@getify/proper-arrows",
    "@getify/proper-ternary",
    "@jm/no-optional-call",
    "import",
    "unicorn",
    "eslint-plugin-no-only-tests",
    "jsdoc",
  ],
  "reportUnusedDisableDirectives": true,
  "rules": {
    "@getify/proper-arrows/name": "error",
    "@getify/proper-arrows/params": [
      "error",
      {
        "allowed": [
          "e",
          "v",
          "cb",
          "fn",
          "pr",
        ],
        "count": 2,
        "length": 3,
        "unused": "trailing",
      },
    ],
    "@getify/proper-arrows/return": [
      "error",
      {
        "ternary": 1,
      },
    ],
    "@getify/proper-arrows/this": [
      "error",
      "nested",
      {
        "no-global": true,
      },
    ],
    "@getify/proper-arrows/where": "error",
    "@getify/proper-ternary/nested": [
      "error",
      {
        "depth": 10,
        "else": true,
      },
    ],
    "@getify/proper-ternary/parens": [
      "error",
      {
        "call": false,
        "object": false,
        "ternary": false,
      },
    ],
    "@getify/proper-ternary/where": [
      "error",
      {
        "return": false,
      },
    ],
    "@jm/no-optional-call/no-optional-call": "error",
    "accessor-pairs": [
      "error",
      {
        "enforceForClassMembers": true,
        "getWithoutSet": false,
        "setWithoutGet": true,
      },
    ],
    "array-callback-return": [
      "error",
      {
        "allowImplicit": false,
        "checkForEach": true,
      },
    ],
    "consistent-return": "error",
    "constructor-super": "error",
    "curly": [
      "error",
      "all",
    ],
    "default-case-last": "error",
    "dot-notation": [
      "error",
      {
        "allowKeywords": true,
      },
    ],
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore",
      },
    ],
    "for-direction": "error",
    "getter-return": [
      "error",
      {
        "allowImplicit": false,
      },
    ],
    "grouped-accessor-pairs": [
      "error",
      "getBeforeSet",
    ],
    "import/default": "error",
    "import/export": "error",
    "import/named": [
      "error",
      {
        "commonjs": true,
      },
    ],
    "import/namespace": "error",
    "import/no-absolute-path": "error",
    "import/no-cycle": [
      "error",
      {
        "commonjs": true,
      },
    ],
    "import/no-deprecated": "error",
    "import/no-dynamic-require": [
      "error",
      {
        "esmodule": true,
      },
    ],
    "import/no-empty-named-blocks": "error",
    "import/no-mutable-exports": "error",
    "import/no-named-as-default": "error",
    "import/no-named-as-default-member": "error",
    "import/no-relative-packages": "error",
    "import/no-self-import": "error",
    "import/no-unresolved": [
      "error",
      {
        "caseSensitiveStrict": true,
        "commonjs": true,
      },
    ],
    "import/no-unused-modules": [
      "error",
      {
        "missingExports": false,
        "unusedExports": true,
      },
    ],
    "import/no-useless-path-segments": [
      "error",
      {
        "commonjs": true,
      },
    ],
    "import/no-webpack-loader-syntax": "error",
    "jsdoc/check-access": "warn",
    "jsdoc/check-alignment": "warn",
    "jsdoc/check-examples": "off",
    "jsdoc/check-param-names": "warn",
    "jsdoc/check-property-names": "warn",
    "jsdoc/check-syntax": "off",
    "jsdoc/check-tag-names": "warn",
    "jsdoc/check-types": "warn",
    "jsdoc/check-values": "warn",
    "jsdoc/empty-tags": "warn",
    "jsdoc/implements-on-classes": "warn",
    "jsdoc/informative-docs": "off",
    "jsdoc/match-description": "off",
    "jsdoc/no-bad-blocks": "warn",
    "jsdoc/no-blank-block-descriptions": "off",
    "jsdoc/no-defaults": "off",
    "jsdoc/no-missing-syntax": "off",
    "jsdoc/no-restricted-syntax": "off",
    "jsdoc/no-types": "off",
    "jsdoc/no-undefined-types": "warn",
    "jsdoc/require-description": "off",
    "jsdoc/require-description-complete-sentence": "off",
    "jsdoc/require-example": "off",
    "jsdoc/require-file-overview": "off",
    "jsdoc/require-jsdoc": "warn",
    "jsdoc/require-param": "warn",
    "jsdoc/require-param-description": "warn",
    "jsdoc/require-param-name": "warn",
    "jsdoc/require-param-type": "warn",
    "jsdoc/require-property": "warn",
    "jsdoc/require-property-description": "warn",
    "jsdoc/require-property-name": "warn",
    "jsdoc/require-property-type": "warn",
    "jsdoc/require-returns": "warn",
    "jsdoc/require-returns-check": "warn",
    "jsdoc/require-returns-description": "warn",
    "jsdoc/require-returns-type": "warn",
    "jsdoc/require-throws": "off",
    "jsdoc/require-yields": "warn",
    "jsdoc/require-yields-check": "warn",
    "jsdoc/valid-types": "warn",
    "new-cap": [
      "error",
      {
        "capIsNew": false,
        "capIsNewExceptions": [
          "Immutable.Map",
          "Immutable.Set",
          "Immutable.List",
        ],
        "newIsCap": true,
        "newIsCapExceptions": [],
      },
    ],
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "off",
    "no-caller": "error",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": [
      "error",
      "always",
    ],
    "no-confusing-arrow": [
      "error",
      {
        "allowParens": false,
      },
    ],
    "no-console": "warn",
    "no-const-assign": "error",
    "no-constant-binary-expression": "error",
    "no-constant-condition": [
      "error",
      {
        "checkLoops": false,
      },
    ],
    "no-constructor-return": "error",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": [
      "error",
      {
        "allowElseIf": false,
      },
    ],
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-pattern": "error",
    "no-eval": "error",
    "no-ex-assign": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-boolean-cast": [
      "error",
      {
        "enforceForLogicalOperands": true,
      },
    ],
    "no-extra-label": "error",
    "no-extra-semi": "off",
    "no-fallthrough": "error",
    "no-func-assign": "error",
    "no-global-assign": [
      "error",
      {
        "exceptions": [],
      },
    ],
    "no-implied-eval": "error",
    "no-import-assign": "error",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-irregular-whitespace": [
      "error",
      {
        "skipStrings": false,
      },
    ],
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": [
      "error",
      {
        "allowLoop": false,
        "allowSwitch": false,
      },
    ],
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-loss-of-precision": "error",
    "no-misleading-character-class": "error",
    "no-mixed-operators": [
      "error",
      {
        "allowSamePrecedence": false,
        "groups": [
          [
            "%",
            "**",
          ],
          [
            "%",
            "+",
          ],
          [
            "%",
            "-",
          ],
          [
            "%",
            "*",
          ],
          [
            "%",
            "/",
          ],
          [
            "/",
            "*",
          ],
          [
            "&",
            "|",
            "<<",
            ">>",
            ">>>",
          ],
          [
            "==",
            "!=",
            "===",
            "!==",
          ],
          [
            "&&",
            "||",
          ],
        ],
      },
    ],
    "no-mixed-spaces-and-tabs": "off",
    "no-multi-str": "error",
    "no-nested-ternary": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-native-nonconstructor": "error",
    "no-new-object": "error",
    "no-new-symbol": "error",
    "no-new-wrappers": "error",
    "no-nonoctal-decimal-escape": "error",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "error",
    "no-only-tests/no-only-tests": [
      "error",
      {
        "block": [
          "And",
          "context",
          "describe",
          "Feature",
          "fixture",
          "Given",
          "it",
          "Scenario",
          "serial",
          "tape",
          "test",
          "Then",
          "When",
        ],
        "focus": [
          "only",
        ],
      },
    ],
    "no-promise-executor-return": "error",
    "no-proto": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-exports": [
      "error",
      {
        "restrictedNamedExports": [
          "default",
          "then",
        ],
      },
    ],
    "no-restricted-globals": [
      "error",
      {
        "message": "Use \`Number.isFinite(..)\` instead, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite#difference_between_number.isfinite_and_global_isfinite",
        "name": "isFinite",
      },
      {
        "message": "Use \`Number.isNaN(..)\` instead, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#difference_between_number.isnan_and_global_isnan",
        "name": "isNaN",
      },
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "defaultstatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top",
    ],
    "no-restricted-syntax": [
      "error",
      {
        "message": "Use \`typeof\` rather than \`instanceof Object\`.",
        "selector": "BinaryExpression[operator='instanceof'][right.name='Object']",
      },
    ],
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-assign": [
      "error",
      {
        "props": true,
      },
    ],
    "no-self-compare": "error",
    "no-sequences": [
      "error",
      {
        "allowInParentheses": false,
      },
    ],
    "no-setter-return": "error",
    "no-shadow": [
      "error",
      {
        "allow": [
          "defaultStatus",
          "event",
          "find",
          "length",
          "name",
          "status",
        ],
        "builtinGlobals": true,
        "hoist": "functions",
        "ignoreOnInitialization": false,
      },
    ],
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "error",
    "no-template-curly-in-string": "error",
    "no-this-before-super": "error",
    "no-throw-literal": "error",
    "no-undef": [
      "error",
      {
        "typeof": true,
      },
    ],
    "no-undef-init": "error",
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": [
      "error",
      {
        "defaultAssignment": false,
      },
    ],
    "no-unreachable": "error",
    "no-unreachable-loop": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": [
      "error",
      {
        "enforceForOrderingRelations": true,
      },
    ],
    "no-unsafe-optional-chaining": [
      "error",
      {
        "disallowArithmeticOperators": true,
      },
    ],
    "no-unused-expressions": [
      "error",
      {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
        "enforceForJSX": true,
      },
    ],
    "no-unused-labels": "error",
    "no-unused-private-class-members": "error",
    "no-unused-vars": [
      "error",
      {
        "args": "after-used",
        "argsIgnorePattern": "^ignore$",
        "caughtErrors": "all",
        "vars": "all",
        "varsIgnorePattern": "^ignore$",
      },
    ],
    "no-use-before-define": [
      "error",
      {
        "allowNamedExports": false,
        "classes": true,
        "functions": true,
        "variables": true,
      },
    ],
    "no-useless-backreference": "error",
    "no-useless-call": "error",
    "no-useless-catch": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-escape": "error",
    "no-useless-rename": [
      "error",
      {
        "ignoreDestructuring": false,
        "ignoreExport": false,
        "ignoreImport": false,
      },
    ],
    "no-useless-return": "error",
    "no-var": "error",
    "no-with": "error",
    "object-shorthand": "error",
    "operator-assignment": [
      "error",
      "always",
    ],
    "prefer-arrow-callback": [
      "error",
      {
        "allowNamedFunctions": false,
        "allowUnboundThis": true,
      },
    ],
    "prefer-const": [
      "error",
      {
        "destructuring": "any",
        "ignoreReadBeforeAssign": true,
      },
    ],
    "prefer-destructuring": [
      "error",
      {
        "AssignmentExpression": {
          "array": true,
          "object": false,
        },
        "VariableDeclarator": {
          "array": false,
          "object": true,
        },
      },
      {
        "enforceForRenamedProperties": false,
      },
    ],
    "prefer-exponentiation-operator": "error",
    "prefer-numeric-literals": "error",
    "prefer-object-spread": "error",
    "prefer-promise-reject-errors": [
      "error",
      {
        "allowEmptyReject": false,
      },
    ],
    "prefer-regex-literals": [
      "error",
      {
        "disallowRedundantWrapping": true,
      },
    ],
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "error",
    "require-atomic-updates": "warn",
    "require-yield": "error",
    "strict": [
      "error",
      "safe",
    ],
    "unicorn/better-regex": "error",
    "unicorn/catch-error-name": "error",
    "unicorn/consistent-destructuring": "error",
    "unicorn/custom-error-definition": "error",
    "unicorn/error-message": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/new-for-builtins": "error",
    "unicorn/no-abusive-eslint-disable": "error",
    "unicorn/no-array-callback-reference": "off",
    "unicorn/no-array-for-each": "off",
    "unicorn/no-array-method-this-argument": "error",
    "unicorn/no-array-push-push": "error",
    "unicorn/no-await-expression-member": "error",
    "unicorn/no-empty-file": "error",
    "unicorn/no-for-loop": "error",
    "unicorn/no-hex-escape": "error",
    "unicorn/no-instanceof-array": "error",
    "unicorn/no-keyword-prefix": "off",
    "unicorn/no-lonely-if": "error",
    "unicorn/no-negated-condition": "error",
    "unicorn/no-nested-ternary": "off",
    "unicorn/no-new-array": "error",
    "unicorn/no-null": "error",
    "unicorn/no-object-as-default-parameter": "error",
    "unicorn/no-static-only-class": "error",
    "unicorn/no-thenable": "error",
    "unicorn/no-this-assignment": "error",
    "unicorn/no-typeof-undefined": "error",
    "unicorn/no-unnecessary-await": "error",
    "unicorn/no-unreadable-array-destructuring": "error",
    "unicorn/no-unreadable-iife": "error",
    "unicorn/no-unsafe-regex": "error",
    "unicorn/no-unused-properties": "error",
    "unicorn/no-useless-fallback-in-spread": "error",
    "unicorn/no-useless-length-check": "error",
    "unicorn/no-useless-promise-resolve-reject": "error",
    "unicorn/no-useless-spread": "error",
    "unicorn/no-useless-switch-case": "error",
    "unicorn/no-useless-undefined": "off",
    "unicorn/no-zero-fractions": "error",
    "unicorn/prefer-array-find": [
      "error",
      {
        "checkFromLast": true,
      },
    ],
    "unicorn/prefer-array-flat": "error",
    "unicorn/prefer-array-flat-map": "error",
    "unicorn/prefer-array-index-of": "error",
    "unicorn/prefer-array-some": "error",
    "unicorn/prefer-at": "off",
    "unicorn/prefer-code-point": "error",
    "unicorn/prefer-date-now": "error",
    "unicorn/prefer-default-parameters": "error",
    "unicorn/prefer-export-from": [
      "error",
      {
        "ignoreUsedVariables": true,
      },
    ],
    "unicorn/prefer-includes": "error",
    "unicorn/prefer-logical-operator-over-ternary": "error",
    "unicorn/prefer-math-trunc": "error",
    "unicorn/prefer-modern-math-apis": "error",
    "unicorn/prefer-module": "off",
    "unicorn/prefer-native-coercion-functions": "error",
    "unicorn/prefer-negative-index": "error",
    "unicorn/prefer-number-properties": "error",
    "unicorn/prefer-object-from-entries": [
      "error",
      {
        "functions": [
          "fromPairs",
          "R.fromPairs",
        ],
      },
    ],
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-prototype-methods": "error",
    "unicorn/prefer-reflect-apply": "error",
    "unicorn/prefer-regexp-test": "error",
    "unicorn/prefer-set-has": "off",
    "unicorn/prefer-set-size": "error",
    "unicorn/prefer-spread": "off",
    "unicorn/prefer-string-replace-all": "error",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-string-starts-ends-with": "error",
    "unicorn/prefer-string-trim-start-end": "error",
    "unicorn/prefer-switch": "error",
    "unicorn/prefer-ternary": "error",
    "unicorn/prefer-top-level-await": "error",
    "unicorn/prefer-type-error": "error",
    "unicorn/relative-url-style": "error",
    "unicorn/require-array-join-separator": "error",
    "unicorn/require-number-to-fixed-digits-argument": "error",
    "unicorn/throw-new-error": "error",
    "use-isnan": [
      "error",
      {
        "enforceForIndexOf": true,
        "enforceForSwitchCase": true,
      },
    ],
    "valid-typeof": [
      "error",
      {
        "requireStringLiterals": true,
      },
    ],
    "yoda": [
      "error",
      "never",
      {
        "onlyEquality": true,
      },
    ],
  },
}
`;
